### ISSUE TITLE
Tab completing expressions deletes the whole line without output

### ISSUE THREAD

**Issue by jglovier** [emmetio/emmet-atom#347]
I've seen several issues that seem similar, but not exactly the same so opening a new one. The problem I'm running into is that when I tab complete an Emmet expression, the whole line disappears without output. 

Observe and behold: 
![emmet-atom-issue](https://cloud.githubusercontent.com/assets/1319791/10898345/853421f2-8197-11e5-8379-f9b124cfdd05.gif)

I've seen others mention that `autocomplete-plus` hijacks Emmet's `tab` shortcut, but as you can see in the gif keybinding resolver shows that Emmet is indeed being fired on `tab`, so I'm at a loss for what is going on here.

What is the solution? Is this just a settings thing?

**Comment by jglovier** [comment]
Possibly related: #225, #282, #334.

**Comment by emilMircea** [comment]
I have the same exact issue! Could not find an adequate solution yet.

**Comment by sergeche** [comment]
Can you open DevTools and check if there are any errors?

**Comment by jglovier** [comment]
@sergeche nothing that appears related to Emmet.

![image](https://cloud.githubusercontent.com/assets/1319791/10914593/2d1cc692-8221-11e5-85e9-5f85119c3f4e.png)

**Comment by sergeche** [comment]
Can you provide a full code (or least code where error can be reproduced) and your current document syntax?

**Comment by jglovier** [comment]
@sergeche I've tried it in several files, and anything with HTML syntax does the same thing.

**Comment by sergeche** [comment]
One possible reason is that if you use some HTML syntax variation (like `HTML (Twig)` or something), Emmet may mistakenly pick wrong settings and output empty result.

Can you try with clean and native `HTML` or `XML` syntax?

**Comment by jglovier** [comment]
Yup, several of these I tried were just plain `.html` files (HTML email files to be exact). Others were `.liquid` files set to HTML syntax.

**Comment by mgravina1** [comment]
I don't know if this can help you.
Go to File->Settings and select "Keybindings". Searching for "tab" you should get all the possible behaviours involving "tab" key. You should also see a message like "You can override these keybindings by copying and pasting them into your keymap file". 
By clicking on the link in this message, a css will be opened. 
Just add the behaviours you want to use at the end of the file (copy them from the previous list).
 
For example:
'atom-text-editor:not([mini])':
  'tab': 'emmet:expand-abbreviation'
  'tab': 'editor:indent'

this will  let you expand emmet abbreviation and indent your code by pressing tab.
Hope it helps

**Comment by calmyournerves** [comment]
This issue bugged me for a long time and I just saw #312. Re-enabling the `snippets` package solved the issue for me.
